### PR TITLE
WT-8767 Return rollback instead of cache full after cache_max_wait timer expires

### PR DESCRIPTION
**Pull request by sulabhM**

### CODE REVIEW

**Review by keithbostic — APPROVED** [review]
This seems fine to me. There was a `test_txn27` failure in the original PR run, but I built locally and wasn't able to reproduce it.

A couple of things I wasn't sure of:

* Why do we decrement `cache->evict_aggressive_score` the first rollback clause (around line 2380), but not in the second one you added? @agorrod, do you know the answer to this question?

* I'd suggest this change, because I think it cleans up the messaging (it's unclear to me why the logged messages would be different, so I made them the same):

```
index a7d8772df..4b995b835 100644
--- a/src/evict/evict_lru.c
+++ b/src/evict/evict_lru.c
@@ -2384,10 +2384,10 @@ __wt_cache_eviction_worker(WT_SESSION_IMPL *session, bool busy, bool readonly, d
         if (!F_ISSET(conn, WT_CONN_RECOVERING) && __wt_cache_stuck(session)) {
             ret = __wt_txn_is_blocking(session);
             if (ret == WT_ROLLBACK) {
-                __wt_verbose_debug(
-                  session, WT_VERB_TRANSACTION, "Rollback reason: %s", "Cache full");
                 --cache->evict_aggressive_score;
                 WT_STAT_CONN_INCR(session, txn_fail_cache);
+                __wt_verbose_debug(
+                  session, WT_VERB_TRANSACTION, "%s", session->txn->rollback_reason);
                 if (app_thread)
                     __wt_verbose_notice(
                       session, WT_VERB_TRANSACTION, "%s", session->txn->rollback_reason);
```

File: test/suite/test_txn27.py [hunk 1]
```diff
@@ -84,6 +84,31 @@ def test_rollback_reason(self):
...
         self.assertEquals(session1.get_rollback_reason(), msg2)
+        session1.rollback_transaction()
+
+        # Fill the cache with updates, applications should be forced into eviction when eviction
+        # updates trigger limit is reached. Keep filling the cache past that point.
+        session1.begin_transaction()
+        for i in range(1, 200):
+            cursor1.set_key(ds.key(i))
+            cursor1.set_value("a"*1024)
+            self.assertEqual(0, cursor1.insert())
```
[the remark this answers is not in the document]
Comment (sulabhM) [reply test/suite/test_txn27.py]: This was just my copy paste, we do not need an assert.

File: src/include/txn.h [hunk 2]
```diff
@@ -19,7 +19,10 @@
  * the session get rollback reason API call. Users of the API could have a dependency on the format
  * of these messages so changing them must be done with care.
  */
-#define WT_TXN_ROLLBACK_REASON_CACHE "oldest pinned transaction ID rolled back for eviction"
+#define WT_TXN_ROLLBACK_REASON_OLDEST_FOR_EVICTION \
+    "oldest pinned transaction ID rolled back for eviction"
+#define WT_TXN_ROLLBACK_REASON_CACHE_FULL \
+    "transaction rolled back otherwise operation would overflow cache"
```
[the remark this answers is not in the document]
Comment (sulabhM) [reply src/include/txn.h]: Done.

**Comment by sulabhM** [comment]
This is still failing the test on Windows for some reason. I am still debugging. To repro, if we run python test multithreaded (even if we run this single test), it fails.

**Comment by agorrod** [comment]
@sulabhM - where are we up to with this PR? It looks like the pending SERVER change has been merged, is it time to resuscitate this one?

**Comment by sulabhM** [comment]
@agorrod, this change was failing testing on Windows for some reason, then I got busy while the ticket was blocked on the SERVER ticket. Yep, time to have another look at this one, I will begin doing so.

**Comment by sulabhM** [comment]
For some weird reason, txn27 keeps failing on this PR. If I run locally on a spawned windows host it doesn't. The failure is complaining about a message on stdout, but that is expected. The message is there on Windows when run on the spawned host but the python runner doesn't fail then.

**Comment by sulabhM** [comment]
@agorrod , this is ready for a review

**Comment by agorrod** [comment]
This change looks good to me.
I'm on my phone so I can't find the proper review button.
I don't remember the details of decrementing the aggressive score. I'm surprised we do it after a failed eviction, but I guess here we've done some work to allow the cache to become unblocked. I'm with Keith, that it makes sense to be consistent.

**Pull request merged**